packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -0,0 +1,107 @@
+import { logger } from '@sentry/utils';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { isESM } from '../../utils/isESM';
+import { findDeclarations, findExports, findIdentifiers, makeAST, removeComments } from './ast';
+import { LoaderThis } from './types';
+const DATA_FETCHING_FUNCTIONS = ['getServerSideProps', 'getStaticProps', 'getStaticPaths'];
+
+type LoaderOptions = {
+  projectDir: string;
+};
+
+/**
+ * Find any data-fetching functions the user's code contains and rename them to prevent clashes, then whittle the
+ * template exporting wrapped versions instead down to only the functions found.
+ *
+ * @param userCode The source code of the current page file
+ * @param templateCode The source code of the full template, including all functions
+ * @param filepath The path to the current pagefile, within the project directory
+ * @returns A tuple of modified user and template code
+ */
+function wrapFunctions(userCode: string, templateCode: string, filepath: string): string[] {
+  let userAST, templateAST;
+  const isTS = new RegExp('\\.tsx?$').test(filepath);
+
+  try {
+    userAST = makeAST(userCode, isTS);
+    templateAST = makeAST(templateCode, false);
+  } catch (err) {
+    logger.warn(`Couldn't add Sentry to ${filepath} because there was a parsing error: ${err}`);
+  }
+
+  // This will be true iff `makeAST` errored, but TS can't quite make the logical leap, so we check it directly
+  if (!userAST || !templateAST) {
+    // Replace the template code with an empty string, so in the end the user code is untouched
+    return [userCode, ''];
+  }
+
+  // Comments are useful to have in the template for anyone reading it, but don't make sense to be injected into user
+  // code, because they're about the template-i-ness of the template, not the code itself
+  removeComments(templateAST);
+
+  for (const fnName of DATA_FETCHING_FUNCTIONS) {
+    const matchingNodes = findIdentifiers(userAST, fnName);
+
+    // If the current function exists in a user's code, prefix all references to it with an underscore, so as not to
+    // conflict with the wrapped version we're going to create
+    if (matchingNodes.length > 0) {
+      matchingNodes.forEach(nodePath => (nodePath.node.name = `_${fnName}`));
```

Comment:
I don't want to stick "sentry" in the name, since the function in question is still their original one, not the sentrified one. It is a good thought, though, to check if they already have something of that name and compensate for it if so. It honestly didn't occur to me that someone might do that, but you're right, there's no reason they couldn't. Good catch!